cmd/tracee-ebpf/internal/printer/printer.go

```diff
@@ -195,15 +178,11 @@ func (p *templateEventPrinter) Init() error {
 
 func (p templateEventPrinter) Preamble() {}
 
-func (p templateEventPrinter) Error(err error) {
-	fmt.Fprintf(p.err, "%v\n", err)
-}
-
 func (p templateEventPrinter) Print(event trace.Event) {
 	if p.templateObj != nil {
 		err := (*p.templateObj).Execute(p.out, event)
 		if err != nil {
-			p.Error(err)
+			logger.Error("error executing tempalte", err)
```

Comment:
But we have to standardize it for our own good.

Reply:
I'll fix it, but I must say it's a bit unintuitive to work like this with implicit key values, since the usage of the function does not imply it at all.

Reply:
It's the way structured logs like zap work.

But logrus has a different API:

![image](https://user-images.githubusercontent.com/3372117/204846251-043c4cd4-e0d7-4d0f-9510-059b3255bdd8.png)

---

Anyway our API could easily be changed to:

```go
logger.Error("executing template", err)

...

func Error(msg string, err error, keysAndValues ...interface{}) {
	errorw(1, pkgLogger, fmt.Sprintf("%s - %s", msg, err.Error()), keysAndValues...)
}
```

In this case, if there's no error, we must use logger.Info()/Warn() etc.

Reply:
It (typed pair) was initially proposed but the use of unspecified types (through interfaces) was picked in the merged PR. I think that were comments asking to make the API cleaner (simple).

> Anyway I fixed the logs to align with it.

👍🏼

Reply:
@NDStrahilevitz Do you think it's worth opening a discussion about the logger api?